cloudinit/config/tests/test_mounts.py

```diff
@@ -0,0 +1,22 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+from unittest import mock
+
+from cloudinit.config.cc_mounts import create_swapfile
+
+
+M_PATH = 'cloudinit.config.cc_mounts.'
+
+
+class TestCreateSwapfile:
+
+    @mock.patch(M_PATH + 'util.subp')
+    def test_happy_path(self, m_subp, tmpdir):
+        swap_file = tmpdir.join("swap-file")
+        fname = str(swap_file)
```

Comment:
isn't this just swap_file.strpath?

Reply:
We had to use `swap_file.strpath` in ubuntu-advantage-tools because `str(swap_file)` didn't work in the version of pytest in trusty. However, using `str()` is preferable because it works against more types, so if `swap_file` changes type for whatever reason then this line doesn't need changing.

(And there is a slight practical component to this: more recent versions of pytest have introduced the `tmp_path` fixture which returns an instance of the stdlib's `pathlib.Path` instead of the third-party-and-in-maintenance-mode `py` library's `py.path.LocalPath`. `str(tmp_path.joinpath(...))` behaves the same as `str(tmpdir.join(...))`, but `tmp_path.joinpath(...).strpath` raises an AttributeError. Eventually (after we're no longer backporting to bionic, now that I look at the version `tmp_path` was introduced in), we'll want to migrate to `tmp_path`.)